network_simulator/host.py

```diff
@@ -23,4 +27,65 @@ def __init__(self, address=None, flows={}, flow_packets_received={},
         self.flow_packets_received = flow_packets_received
         self.link = link
 
-    # TODO(team): More event subclasses to come?
\ No newline at end of file
+
+class HostReceivedPacketEvent(Event):
+    """
+    Host receives packets from a flow.
+    """
+
+    def __init__(self, host, packet):
+        """
+        Sanity check: make sure destination was this Host and that this isn't
+        a RouterPacket.
+        :ivar host: host that receives packet
+        :ivar packet: packet received
+        """
+        Event.__init__(self)
+
+        # Sanity check
+        assert packet.dest_id == host and \
+        packet.packet_id != ROUTER_PACKET_DEFAULT_ID
```

Comment:
The check should be:

assert packet.dest_id == host.address and not isinstance(packet, RouterPacket)

Since dest_id is a string and host is a Host, you cannot directly compare them. Also, stylistically, checking if the Packet is an instanceof a RouterPacket is cleaner.

Also, when you continue the same "line" of code over multiple lines with a \, indent the second line by one tab. This is part of the PEP8 style guidelines. PyCharm should automatically do this though.